src/probnum/kernels/_white_noise.py

```diff
@@ -0,0 +1,52 @@
+"""White noise kernel."""
+
+from typing import Optional
+
+import numpy as np
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class WhiteNoise(Kernel[_InputType]):
```

Comment:
is there a use case for the white noise kernel? I get all the others, but am not sure whether `WhiteNoise` is useful for e.g. Bayesian quadrature.

Reply:
Once kernel arithmetic is added one may want to define a sum kernel with a white noise kernel. Take Bayesian optimization for example. If you assume your function can only be evaluated up to noise, then this corresponds to a Gaussian likelihood or equivalently a sum kernel with a White noise component and an exact likelihood.

Reply:
I was wondering less about future applications, but more so about immediate ones. Isn't BO out of scope for the foreseeable future? 
>  If you assume your function can only be evaluated up to noise, then this corresponds to a Gaussian likelihood or equivalently a sum kernel with a White noise component and an exact likelihood.

Is this the way we would model such noisy evaluations though? 

Just to be clear: I do not disagree with having a white noise kernel and also not with the way it is implemented! I was just wondering whether it has a purpose in the near future or not (we will have to maintain the code if it is there, so it better be useful). If you want it in, we put it in!